src/main/kotlin/org/rust/lang/core/macros/MacroExpansion.kt

```diff
@@ -8,37 +8,126 @@ package org.rust.lang.core.macros
 import com.intellij.psi.util.CachedValueProvider
 import com.intellij.psi.util.PsiModificationTracker
 import org.rust.cargo.project.settings.rustSettings
-import org.rust.lang.core.psi.RsMacro
-import org.rust.lang.core.psi.RsMacroCall
+import org.rust.lang.core.psi.*
 import org.rust.lang.core.psi.ext.RsElement
+import org.rust.lang.core.psi.ext.childrenOfType
+import org.rust.lang.core.psi.ext.descendantOfTypeStrict
 import org.rust.lang.core.psi.ext.macroName
-import org.rust.lang.core.psi.rustStructureModificationTracker
 
-private val NULL_RESULT: CachedValueProvider.Result<List<RsExpandedElement>?> =
+private val NULL_RESULT: CachedValueProvider.Result<MacroExpansion?> =
     CachedValueProvider.Result.create(null, PsiModificationTracker.MODIFICATION_COUNT)
 
-fun expandMacro(call: RsMacroCall): CachedValueProvider.Result<List<RsExpandedElement>?> {
+fun expandMacro(call: RsMacroCall): CachedValueProvider.Result<MacroExpansion?> {
```

Comment:
Maybe convert it into an extension function of `RsMacroCall` not to provide additional name into global scope? Same comment about other public functions in `MacroExpansion.kt` and `MacroExpansionViewUtils.kt`.

Reply:
This will be highly refactored with #3015, so I don't think it is not necessary